rasa/nlu/classifiers/diet_classifier.py

```diff
@@ -242,6 +246,10 @@ def __init__(
 
         self.data_example = None
 
+        self.label_key = (
+            "label_ids" if self.component_config[INTENT_CLASSIFICATION] else "tag_ids"
```

Comment:
Can `tag_ids` be used as `self.label_key`?

Reply:
Yes, however we don't balance the data in that case. But if we do not update the key, we will get an error if DIETClassifier is just used as entity extractor

Reply:
I think ModelData can get `label_key=None`
however let's keep non balancing for sequence labels (for core policy)

Reply:
If `label_key=None` we will currently raise an error. Also, just saw that `split` should also not work if `label_key` is set to `tag_ids`. Will fix that.

Reply:
why don't we remove the error? 😅

Reply:
I think, it'd make the most sense to set it to `None`.

Reply:
agree, already working on it, just need to update the `split` method

Reply:
could you please set `label_key` to `None` in case of full dialogue featurizer in ted as well